### DISCUSSION TITLE
Folder Templates, Obsidian Sync and Obsidian being open on two Machines

### DISCUSSION THREAD

**Discussion by finn-matti** [SilentVoid13/Templater#1096] [Help] [answered]
(I thougt I'd start this as a discussion first, before filing an issue. Maybe there is a solution?!)

My Setup is as follows:

- I have two computers that are almost always running and have Obsidian opened
- I have Obsidian Sync activated and it syncs not only my notes but my plugins and settings as well
- I have set a folder template for my 'inbox' folder

The problem I run into:

When creating a new note in my inbox folder the **folder template gets triggered twice**:

- once on the currently used machine
- and after a sync back and forth on the currently unused machine

The reason, I Suspect, is that the file, upon being created, gets synced as an empty file to the unused machine, where it triggers the folder template, while at same time on the currently used machine it also triggers the folder template. When the file is being synced back to the currently used machine, Obsidian merges the files - but since the templates include a timestamp - they are both not exactly the same leading to a duplication of the templates content.

Is there a way around that? I would assume that if Obsidian sync would wait slightly longer, that it would sync the contents of the file, therefore leading to the desired outcome.

Another kinda funny workaround is to create the file let it sync back and forth and let the idle machine apply the template to the new file. But that is kinda slow and it happens that only of the machines is actually running.

What to do? 😅

**Comment by Zachatoo** [comment]
Some options I can think of:

1. Don't have Obsidian open on more than one device. Templater is not the only plugin that this could potentially cause issues with, any plugin that can modify files soon after creation will cause this same issue (QuickAdd is another that would likely have this same issue).
2. Have a different config per device, and only have one of your devices setup to use folder templates. You can do this in `Settings - About - Override config folder`. If only one device is watching your filesystem for new files and immediately updating them, then there won't be conflicts.
3. Pause sync, create the file, then unpause sync. Not sure if this will work, but it might.

There is no way to tell Obsidian Sync to wait longer unfortunately. There's not an easy way to fix this in Templater since Templater is listening to the "file created" event, the file has already been created and started syncing before Templater can do anything to stop it from syncing, assuming there was a way to prevent a file from syncing.

TLDR this is a symptom of how Obsidian Sync works, and I don't think there's anything that can be done on the Templater side to resolve this.

**Comment by finn-matti** [comment]
@Zachatoo Yeah, that's what I thought. I would not be against closing Obsidian on the inactive computer, but it's error prone, because there is a human involved. There really should be a plugin or an affordance in Obsidian itself, that differentiates the currently active instance.

Alas, there is no good way to prevent this from happening. Just what I thought. Thanks for thinking it through for me!

**Comment by UsefullPartBrad** [comment, marked as answer]
<%* app.internalPlugins.getPluginById('sync').disable()-%> at the beginning of a template and <%* app.internalPlugins.getPluginById('sync').enable() -%> at the end often intercepts the sync (not 100%) and allows the template to complete before syncing.

**Reply by finn-matti** [reply]
This seems to be working well enough in my limited testing so far. So I marked it as answered. Clever idea!

A couple of points:

- I changed the script so it would not explode when it can't find the plugin with the id sync like so: `<%* app.internalPlugins.getPluginById('sync')?.disable()-%>` and `<%* app.internalPlugins.getPluginById('sync')?.enable()-%>`
- For anybody else being confused about the asterisk in the beginning of the template snippets: This marks the snippet as [a JavaScript execution command](https://silentvoid13.github.io/Templater/commands/overview.html#command-types). I can't seem to find out what the dash in the end of the snippets does, though.

**Reply by UsefullPartBrad** [reply]
an additional thought: If you are using the button plugin to launch the template, use the "templater true" command at the end of the button and have somewhere on the button page <%* app.internalPlugins.getPluginById('sync')?.disable()-%> . The only problem: I have no idea how to get it to restart the sync plugin if the template terminates due to filename conflict. any ideas there would be helpful. I would thing it would be a error catch in the templater file with a sync enable followed by a file self deletion, but I haven't found a way to make that work.

**Reply by nicolnt** [reply]
Since I have implemented this one month ago, the files have been very much less frequently overwritten. So thank you.
Hopefully we'll have a fix that works all the time to prevent content loss.

Here's a bug report in Obsidian forum about this issue: https://forum.obsidian.md/t/obsidian-sync-updates-from-one-device-overwritten-by-mobile-android-device/

**Comment by iamrecursion** [comment]
I would suggest hoisting the sync disable/enable logic into Templater itself.

**Comment by zsviczian** [comment]
I found this solution for myself. You can refine the script to filter out a folder, a path or a filename like in this example. Prevents Templater from cannibalizing templates during sync.

```js
<%*
  if(tp.file.title === "TITLE OF YOUR FILE") {
    throw(new Error("Aborting Templater because it is cannibalizing it's own template (likely due to Obsidian Sync to a new device)"));
  }
%>
```

I have a longer writeup and some additional examples here: https://excalidraw-obsidian.online/Hobbies/Excalidraw+Blog/WIKI/07+Developer+Docs/Obsidian+Templater+Tricks#Templater+Cannibalizes+Templates+with+Obsidian+Sync

**Reply by LeonelRFF** [reply]
I think the link is outdated, but I think it refers to this one: https://excalidraw-obsidian.online/WIKI/07+Developer+Docs/Obsidian+Templater+Tricks#Templater+Cannibalizes+Templates+with+Obsidian+Sync
